### DISCUSSION TITLE
Issues installing on Centos7

### DISCUSSION THREAD

**Discussion by achillean00** [Security-Onion-Solutions/securityonion#3666] [Unsupported Versions] [answered]
Hi,

I ran into a couple of issues install SO on a vanilla Centos7 box so I thought I'd document them.

## Tagged VLAN support
The management interface for this box is on a tagged VLAN. This results in an interface name that looks something like:
`enp49s0f0.1000@enp49s0f0`

When the installer grabs the list of interfaces to build nic_list in so-functions it grabs it as is, unfortunately this isn't really a valid interface name to pass to other commands, such as `ip a s` which it later tries to run. The actual interface name is `enp49s0f0.1000`. This naming convention appears to be standard, and I couldn't get CentOS to call it something else so in the end I edited the following lines in `check_requirements()` in so-functions from
```
readarray -t nic_list <<< "$(ip link| awk -F: '$0 !~ "lo|vir|veth|br|docker|wl|^[^0-9]"{print $2}' | grep -vwe "bond0"  | sed 's/ //g')"
```
to
```
readarray -t nic_list <<< "$(ip link| awk -F: '$0 !~ "lo|vir|veth|br|docker|wl|^[^0-9]"{print $2}' | grep -vwe "bond0"  | sed 's/ //g' | sed 's/@.*$//g')"
```
and also in filter_unused_nics() from
```
filtered_nics=$(ip link | awk -F: '$0 !~ "lo|vir|veth|br|docker|wl|^[^0-9]"{print $2}' | grep -vwe "$grep_string"  | sed 's/ //g')
```
to
```
filtered_nics=$(ip link | awk -F: '$0 !~ "lo|vir|veth|br|docker|wl|^[^0-9]"{print $2}' | grep -vwe "$grep_string"  | sed 's/ //g' | sed 's/@.*$//g')
```
given I've gone for a greedy regular expression this could probably be done in a safer way, but it worked for me.

## Version Mismatch
Part way through the install after copying the public key from the sensor to the master the script checks to see if a different version of so-setup is available on the master and copies it to the sensor to install.
This kept bombing out for me with `bash: /root/manager_setup/securityonion/setup/so-setup: No such file or directory`

Looking at the code it downloads and extracts a tar file and expects to find so-setup in the path above. However the contents of the tar file on my master have the path tmp/sogh/securityonion/ leading to the error. I bodged the line in so-setup from
```
exec bash /root/manager_setup/securityonion/setup/so-setup "${original_args[@]}"
```
to
```
exec bash /root/manager_setup/securityonion/tmp/sogh/securityonion/setup/so-setup "${original_args[@]}"
```
but I guess the path in the tar file is probably what actually needs fixing?

HTH, Ian

**Comment by TOoSmOotH** [comment, marked as answer]
Added #3687 to look at this although it's not high on the priority list. 

A couple of suggestions:

- Change the default VLAN for the switch port to 1000 and use the normal name. 
- If you want to test your fix, replace the tarball with the tarball of your modified code.
- If that works out throw a PR to dev :)
